modeshape-jcr/src/main/java/org/modeshape/jcr/index/local/ManagedLocalIndex.java

```diff
@@ -85,4 +93,17 @@ public void removeAll() {
         index.removeAll();
     }
 
+    @Override
+    public IndexManager.IndexStatus getStatus() {
+        return status.get();
+    }
+
+    @Override
+    public void updateStatus(IndexManager.IndexStatus currentStatus, IndexManager.IndexStatus newStatus) {
+        this.status.compareAndSet(currentStatus, newStatus);
```

Comment:
I'm not sure especially because of the "highly" multi-threaded nature of the indexes:  for example the `enable` & `disable` methods are present on the `ManagedIndex` SPI, but atm. their only use is a limited case of reindexing. By limited case I mean in a very particular scenario (at startup for example) indexes will go through `enabled` -> `disabled` sequences while they're being reindexed.
On the other hand, there is no SPI support for what's happening when `workspace.reindex` is called, even though it's semantically the same thing as above.

Introducing assertions or exceptions in this method would essentially mean "only moving between state A and state B" is correct, anything else is wrong. We could do that, but based on the current existing SPI it's totally unclear to me what those correct transitions are. So the implementation of this method at is "happy case" attempt of certain transitions I know for a fact can occur:
* enabled
* enabled - reindexing
* reindexing - enabled
* enabled - disabled